inst/include/sampler.h

```diff
@@ -457,13 +533,21 @@ namespace smc {
         pAdapt->updateEnd(algParams,pPopulation);
 
         //Finally, the current particle set should be appended to the historical process.
-        if(htHistoryMode != HistoryType::NONE) {
+        if(htHistoryMode != HistoryType::NONE){
             History.clear();
             historyelement<Space> histel;
-            histel.Set(N, pPopulation, nAccepted, nRepeats, historyflags(nResampled));
+            switch(htHistoryMode) {
+            case HistoryType::RAM:
+                histel.Set(N, pPopulation, nAccepted, nRepeats, historyflags(nResampled));
+                break;
+            case HistoryType::AL:
+                histel.Set(N, pPopulation, nAccepted, nRepeats, historyflags(nResampled), uRSIndices);
+            /// To avoid compiler warnings, HistoryType::NONE is handled

```

Comment:
Don't you want a `break` here? Rather than falling through to the NONE case?

Reply:
Done!